jadx-gui/src/main/java/jadx/gui/ui/codearea/FridaAction.java

```diff
@@ -32,157 +32,134 @@
 
 	public FridaAction(CodeArea codeArea) {
 
 		super(NLS.str("popup.frida") + " (f)", codeArea);
-		LOG.info("triggered meee");
 		KeyStroke key = getKeyStroke(KeyEvent.VK_F, 0);
 		codeArea.getInputMap().put(key, "trigger frida");
 		codeArea.getActionMap().put("trigger frida", new AbstractAction() {
 			@Override
 
 			public void actionPerformed(ActionEvent e) {
 				node = getNodeByOffset(codeArea.getWordStart(codeArea.getCaretPosition()));
-				copyFridaCode();
+
+				String fridaSnippet = generateFridaSnippet();
+
+				Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+				StringSelection selection = new StringSelection(fridaSnippet);
+				clipboard.setContents(selection, selection);
 			}
 		});
 	}
 
-	private void copyFridaCode() {
-
-		if (node != null) {
-			if (node instanceof JMethod) {
-				JMethod n = (JMethod) node;
-				MethodNode methodNode = n.getJavaMethod().getMethodNode();
-				MethodInfo mi = methodNode.getMethodInfo();
-				methodName = mi.getName();
-				if (methodName.equals("<init>") || methodName.equals("onCreate")) {
-					methodName = "$init";
-				}
-				String fullClassName = methodNode.getParentClass().getFullName();
-				String className = methodNode.getParentClass().getShortName();
-				LOG.debug("node is jmethod");
-				ClassNode tmp = methodNode.getParentClass();
-				while (true) {
-					if (!tmp.isTopClass()) {
-						fullClassName = fullClassName.substring(0, fullClassName.lastIndexOf(".")) + "$"
-								+ fullClassName.substring(fullClassName.lastIndexOf(".") + 1, fullClassName.length());
-					} else {
-						break;
-					}
-					tmp = tmp.getParentClass();
-				}
-				JMethod jMth = (JMethod) node;
-				int mthLine = jMth.getLine();
-				List<String> argNames = jMth.getRootClass().getCodeInfo().getAnnotations().entrySet().stream()
-						.filter(e -> e.getKey().getLine() == mthLine && e.getValue() instanceof VarDeclareRef)
+	private String generateFridaSnippet() {
+		if (node instanceof JMethod) {
+			LOG.debug("node is jmethod");
+			return generateMethodSnippet((JMethod) node);
+		} else if (node instanceof JClass) {
+			LOG.debug("node is jclass");
+			return generateClassSnippet((JClass) node);
+		} else if (node instanceof JField) {
+			LOG.debug("node is jfield");
+			return generateFieldSnippet((JField) node);
+		}
+		LOG.debug("cannot generate frida snippet from node");
+		return "";
+
+	}
+
+	private String generateMethodSnippet(JMethod jMth) {
+		JavaMethod javaMethod = jMth.getJavaMethod();
+		MethodInfo methodInfo = javaMethod.getMethodNode().getMethodInfo();
+		String methodName = methodInfo.getName();
+		if (methodName.equals("<init>") || methodName.equals("onCreate")) {
+			methodName = "$init";
+		}
+		String rawClassName = javaMethod.getDeclaringClass().getRawName();
+		String shortClassName = javaMethod.getDeclaringClass().getName();
+
+		String functionUntilImplementation;
+		if (!javaMethod.getMethodNode().getOverloads().isEmpty()) {
```

Comment:
```suggestion
		if (javaMethod.getMethodNode().isArgsOverloaded()) {
```
Will this method work here? If not and you still need yours `getOverloads()` method, move it to this class, because java definition of overloaded methods is slightly different (i.e. you don't check args count and inherited methods).

Reply:
I was under the impression that overloads are any method in the same class that has the same name, no matter how much parameters it has, or their types.
So this would be considered an example of overloading:
```java
class Addition {
    public int add(int a, int b)
    {
        int sum = a + b;
        return sum;
    }

    public int add(int a, int b, int c)
    {
        int sum = a + b + c;
        return sum;
    }
}
```
I'm pretty sure that Frida uses this definition when it requires specifying the overload (to know which method, from all of the method with the same name, to call).
I will move the method to this class if you still think it doesn't belong in  `MethodNode`.

I'm not sure what Frida does when there is also an inherited method with the same name. I think it's fine to leave the logic like this and add more sophisticated logic for overloading later, if it's needed.

Reply:
Well, I tried to check [java specs](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.4.9)... and it is very cryptic, so I am not sure about args count match, but examples show inheritance. So it is better to check that on sample code.
Anyway, in jadx I use overload with a same args count, and it will be confusing to keep different implementations for same thing 😕

Reply:
I checked, and Frida only requires `.overload()` when there are methods with the same name in the same class.
It doesn't care about inheritance, but it does care about other methods with different types of arguments.

I moved the overload check to this class to avoid the confusion with the other overload method 😄

Reply:
Great! Thanks!